app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,85 @@
+class PostsController < ApplicationController
+  before_action :find_post, only: [:show, :edit, :update, :destroy]
+  before_action :authenticate_user, except: [:index, :show]
+  before_action :authorize_user, only: [:edit, :update, :destroy]
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new post_params
+    @post.user = current_user
+    if @post.save
+      redirect_to post_path(@post)
+      flash[:notice] = "Post created!"
+    else
+      render :new
+      flash[:alert] = "Post not created!"
+    end
+  end
+
+  def show
+    @comment = Comment.new
+    comments = @post.comments
+    respond_to do |format|
+      format.html { render }
+      format.json { render json: @post}
+    end
+  end
+
+  def index
+    if params[:q]
```

Comment:
"q" is not particularly descriptive. I'm assuming this is a carryover from Awesome Answers (short for "question")?